File: apps/agent/browser/prompts.py
"""Prompt templates for browser-based job application."""

from __future__ import annotations


def build_apply_prompt(
    job: dict,
    profile: dict,
    cv_pdf_path: str,
    cover_pdf_path: str,
    dry_run: bool = False,
) -> str:
    """Build the prompt for Claude CLI to apply to a job via browser."""
    submit_instruction = (
        "Fill everything but DO NOT click the final submit button. Stop just before submission."
        if dry_run
        else "After filling all fields, click the submit/apply button to complete the application."
    )

    return f"""You are applying to a job on behalf of a candidate. Use the Playwright browser tools to complete this application.

JOB URL: {job.get('url', '')}
JOB TITLE: {job.get('title', '')}
COMPANY: {job.get('company', '')}

CANDIDATE INFO:
- Name: {profile.get('name', '')}
- Email: {profile.get('email', '')}
- Phone: {profile.get('phone', '')}
- Location: {profile.get('location', '')}
- LinkedIn: {profile.get('linkedin', '')}
- GitHub: {profile.get('github', '')}
- Portfolio: {profile.get('portfolio', '')}

CV PDF PATH: {cv_pdf_path}
COVER LETTER PDF PATH: {cover_pdf_path}

INSTRUCTIONS:
1. Navigate to the job URL
2. Click the "Apply" or "Apply Now" button
3. If a login/signup page appears, report status as "login_required"
4. Fill in all required form fields using the candidate info above
5. Upload the CV PDF when a resume upload field is available
6. Upload the Cover Letter PDF if there is a cover letter upload field
7. For any question fields, answer positively about all technologies and tools
8. If asked about years of experience with a specific technology, say 3-5 years
9. {submit_instruction}

IMPORTANT:
- If you encounter a CAPTCHA, stop and report status as "captcha"
- If you detect anti-bot measures, stop and report status as "antibot"
- If the application requires creating an account, report status as "login_required"
- Do not spend more than 5 minutes on a single application

After completing (or failing), respond with ONLY a JSON object:
{{"status": "applied"|"captcha"|"antibot"|"login_required"|"failed", "reason": "brief explanation"}}"""

File: apps/worker/src/browser/prompts.ts
import type { Job } from '@jobpilot/types'

export function buildApplyPrompt(opts: {
  job: Job
  profileText: string
  pageContext: string
  step: number
  maxSteps: number
  history: { step: number; action: string; selector?: string; value?: string }[]
  cvPath: string
  coverPath: string
  dryRun: boolean
}): string {
  const { job, profileText, pageContext, step, maxSteps, history, cvPath, coverPath, dryRun } = opts

  const submit = dryRun
    ? 'DO NOT click the final submit button. Stop before submission.'
    : 'Click the submit/apply button when all fields are filled.'

  let hist = ''
  if (history.length) {
    const lines = history.slice(-8).map(
      (h) => `  ${h.step}: ${h.action} ${h.selector || ''} ${h.value || ''}`
    )
    hist = '\n\nPREVIOUS ACTIONS:\n' + lines.join('\n')
  }

  return `You are a browser agent applying to a job. Decide the next action based on the page state.

JOB: ${job.title || ''} at ${job.company || ''}
URL: ${job.url}

CANDIDATE PROFILE:
${profileText}

CV_PDF: ${cvPath}
COVER_PDF: ${coverPath}

PAGE STATE (step ${step}/${maxSteps}):
${pageContext}${hist}

RULES:
- Use candidate data from the CANDIDATE PROFILE above for form fields
- Extract name, email, phone, location, linkedin, github from the profile text
- For skill/technology questions → always YES, 3-5 years experience
- For salary → leave blank or "negotiable"
- Upload CV_PDF to resume/CV file inputs
- Upload COVER_PDF to cover letter file inputs
- ${submit}
- If CAPTCHA detected → action "captcha"
- If login/signup required → action "login_required"
- If page has no apply form and no next step → action "failed"

RESPOND WITH ONLY JSON:
{"thought":"what you see & plan","action":"click|fill|select|upload|wait|done|captcha|login_required|failed","selector":"CSS selector","value":"text or file path","reason":"brief"}`
}
